e2e_testing/torchscript/elementwise.py

```diff
@@ -86,23 +86,27 @@ def ElementwiseTernaryModule_basic(module, tu: TestUtils):
 
 
 class ElementwiseWhereSelfModule(torch.nn.Module):
-    def __init__(self):
-        super().__init__()
 
-    @export
-    @annotate_args([
-        None,
-        ([-1, -1, -1], torch.float32, True),
```

Comment:
why was this changed?

Reply:
`a>0.5` where a is of type float32 whereas 0.5 is to type float64. This is a type mismatch and aten.gt.Scalar would throw an error since type promotion is not handled there.

Reply:
Previously, it wasn't promoting a to match type of (0.5) which is wrong, it should have promoted a.

Reply:
Type promotion is a very confusing and messy thing 😂 when it comes to scalars that are part of a tensor operation, if the scalar is a float, it gets converted to a `float32` (the default float type) and then the promotion takes place between the tensor and the scalar.

https://github.com/pytorch/pytorch/blob/deb6989880d8a62bb45ce6c6b058bb5d2e28cf91/aten/src/ATen/native/TypeProperties.cpp#L98-L105

If the scalar is part of an op that does not involve tensors, such as `NumToTensor` then the scalar does get converted to `float64` if it was a float scalar.

Therefore, for the case where the input tensor is a float tensor, all you have to do is convert the scalar to match the type of the float tensor.

Reply:
In general, it is not a good practice to remove a unit test once the unit test has been added. New changes should not break features that were already supported. I recommend undoing the changes you made for the float tensor types and try to get the lowering to work correctly with them